src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,40 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * one of the Tasks that user's can add into their list
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
+
+    /**
+     * constructor to initialise a Deadline object
+     * @param description the Task description that is obtained from the Task class
+     * @param by the deadline time component that is stored as a LocalDateTime
```

Comment:
I think you should add punctuation behind each parameter description as stated [here](https://se-education.org/guides/conventions/java/intermediate.html#comments). Moreover, there should also be an empty line between description and parameter section.

You can apply these rules for the rest of your JavaDoc!